tests/test_selection/conftest.py

```diff
@@ -55,3 +55,27 @@ def df_test_num_cat():
     X = df[["var_A", "var_B", "var_C", "var_D"]]
     y = df["target"]
     return X, y
+
+
+@pytest.fixture(scope="module", name="dummy_method")
+def wrap_dummy_method():
+    def wrapper():
+        import random
```

Comment:
may I confess that I didn't know this pytest wrapping functionality? would anyone fancy enlightening me? :p

Thank you!

Reply:
The fixture is returning a function, which is then returning another function.

Git is complaining about me checking out this branch (perhaps because it is a PR off a feature branch?), but my initial thoughts were:

- Do we need the triple nesting of functions? 
- Do we need to immediately invoke the returned function?

Reply:
well, here it injects a function to a test through a wrapper. As the output of fixtures can be passed to testing functions, is like a hack to pass a function to a test. Perhaps there's a better way of doing this.

Reply:
@ChristopherGS triple nesting does not look nice, is there a way we can a pass a method to a test ? I think inside test we could import it from some testing_utils.

Reply:
Would something like this do the trick?

```
@pytest.fixture()
def argument_printer():
    def _foo(*args, **kwargs):
        return (args, kwargs)

    return _foo


def test_example(argument_printer):
    first_case = argument_printer('a', 'b')
    assert first_case == (('a', 'b'), {})

    second_case = argument_printer('a', b=2, c='test')
    assert second_case == (('a',), {'b': 2, 'c': 'test'})

```

Reply:
@ChristopherGS yes , thanks for the clarity.